docs/manual/src/udl/custom_types.md

```diff
@@ -1,72 +1,110 @@
 # Custom types
 
-Custom types allow you to extend the UniFFI type system to support types from your Rust crate or 3rd
-party libraries. This works by converting to and from some other UniFFI type to move data across the
-FFI. You must provide a `UniffiCustomTypeConverter` implementation to convert the types.
+Custom types offer a way to use types in your interface that don't derive one of the UniFFI type traits (`uniffi::Record`, `uniffi::Enum`, `uniffi::Object`, etc.).
+Instead, custom types are converted to/from a type that does derive those traits, called a "bridge type", when being passed across the FFI.
+
+Custom types are often used with structs that wraps a builtin type, for example `Guid(String)`.
+These types can be passed across the FFI as the builtin type, which can be more efficient than passing a struct.
+The foreign bindings will treat these types as the builtin type, for example `Guid` could appear a string to the foreign code.
+
+Custom types can also be customized on the foreign side.  For example, a URL could be:
+* Represented by the `url::Url` type in Rust
+* Passed across the FFI as a string
+* Represented by the `java.net.URL` type in Kotlin
 
 ## Custom types in the scaffolding code
 
-Consider the following trivial Rust abstraction for a "handle" which wraps an integer:
+### custom_type!
+
+Use the `custom_type!` macro to define a new custom type.
 
 ```rust
-pub struct Handle(i64);
+
+// Some complex struct that can be serialized/deserialized to a string.
+use some_mod::SerializableStruct;
+
+// When passing `SerializableStruct` objects to the foreign side, they will
+// be converted to a `String`, then `String` will be used as the bridge type
+// to pass the value across the FFI. Conversly, when objects are passed to
```

Comment:
typo: Conversely

also: maybe explicitly note here that this example assumes the relevant Into/TryInto implementations already exist? (it's implied, but maybe worth spelling out)